### PR TITLE
Fix bug: populate cache deps [skip ci]

### PR DESCRIPTION
**Pull request by YanxuanLiu**
fix https://github.com/NVIDIA/spark-rapids/issues/11796

Corrected the argument of `build/get_buildvers.py` to fix bug of getting build versions.

### CODE REVIEW

**Comment by pxLi** [comment]
I think the original ticket https://github.com/NVIDIA/spark-rapids/issues/11796 is also asking for `fix both error masking and the underlying issues`

so we should correctly error out for pipeline usage, please try with `set -o pipefail`

**Comment by YanxuanLiu** [comment]
> I think the original ticket #11796 is also asking for `fix both error masking and the underlying issues`
> 
> so we should correctly error out for pipe usage, please try with `set -o pipefail`

Updated the PR:

1. added `set -o pipefail` to catch errors in pipe
2. split commands to ensure `set -e` can work as expected.


Test build: https://github.com/YanxuanLiu/spark-rapids/actions/runs/12146768557/job/33871438631

File: .github/workflows/mvn-verify-check/populate-daily-cache.sh [hunk 1]
```diff
@@ -14,27 +14,30 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-set -x
-max_retry=3; delay=30; i=1
+set -e
```
Comment (gerashegalov) [inline .github/workflows/mvn-verify-check/populate-daily-cache.sh:17]: would be good to standardize in the future if we enable debug output by default.

**Review by gerashegalov — APPROVED** [review]
LGTM

**Review by NvTimLiu — APPROVED** [review]
LGTM

File: .github/workflows/mvn-verify-check/populate-daily-cache.sh (hunk 1 above)
Comment (YanxuanLiu) [reply .github/workflows/mvn-verify-check/populate-daily-cache.sh:17]: Tested in [FAILED](https://github.com/YanxuanLiu/spark-rapids/actions/runs/12151847128/job/33887174202), the workflow failed with the bug.

Yes, for this script we disable debug output for clear log.

**Comment by YanxuanLiu** [comment]
build

**Review by pxLi — APPROVED** [review]

**Pull request merged**